plotly/tools.py

```diff
@@ -1503,35 +1503,53 @@ def _unconvert_from_RGB_255(colors):
             return un_rgb_colors
 
     @staticmethod
-    def _map_array2color(array, colormap, vmin, vmax):
+    def _map_face2color(face, colormap, vmin, vmax):
         """
-        Normalize values in array by vmin/vmax and return plotly color strings.
+        Normalize facecolor values by vmin/vmax and return rgb-color strings
 
-        This function takes an array of values along with a colormap and a
-        minimum (vmin) and maximum (vmax) range of possible z values for the
-        given parametrized surface. It returns an rgb color based on the
-        relative position of zval between vmin and vmax
+        This function takes a tuple color with elements between 0 and 1, along
+        with a colormap and a minimum (vmin) and maximum (vmax) range of
+        possible mean distances for the given parametrized surface. It returns
+        an rgb color based on the mean distance between vmin and vmax
 
         """
         if vmin >= vmax:
             raise exceptions.PlotlyError("Incorrect relation between vmin "
                                          "and vmax. The vmin value cannot be "
                                          "bigger than or equal to the value "
                                          "of vmax.")
-        # find distance t of zval from vmin to vmax where the distance
-        # is normalized to be between 0 and 1
-        t = (array - vmin) / float((vmax - vmin))
-        t_colors = FigureFactory._find_intermediate_color(colormap[0],
-                                                          colormap[1],
-                                                          t)
-        t_colors = t_colors * 255.
-        labelled_colors = ['rgb(%s, %s, %s)' % (i, j, k)
-                           for i, j, k in t_colors.T]
-        return labelled_colors
+        # find the normalized distance t of a triangle face between vmin and
+        #vmax where the distance is normalized between 0 and 1
+        t = (face - vmin) / float((vmax - vmin))
+
+        if len(colormap) <= 1:
+            t_color = colormap[0]
+            color = FigureFactory._convert_to_RGB_255(t_color)
+            color = FigureFactory._label_rgb(color)
+        else:
+            # account for colormaps with more than one color
+            incr = 1./(len(colormap) - 1)
+            low_color_index = int(t/incr)
+
+            if t == 1:
+                t_color = colormap[low_color_index]
+                color = FigureFactory._convert_to_RGB_255(t_color)
+                color = FigureFactory._label_rgb(color)
+
+            else:
+                t_color = FigureFactory._find_intermediate_color(
```

Comment:
Mk, I'll wait until this is pushed and then I can iterate off of that. Just a thought - this kind of functionality might be useful in the `colorlover` package. It'd be useful if there was a `ColorMapper` class that would do the kinds of interpolation that you're talking about here. That might be a more appropriate place to handle interpolating within or between an arbitrary list of colors. So you could initialize it with a list of colors (like we have here) and an arbitrary number of bins (e.g., `mymap = cl.ColorMapper(color_list, n_bins=1000)`) and then you could pass a number between 0 and 1 to return the color for that bin (e.g.,  `mymap(.2)` and it'd return the index according to int(n_bins * .2)).